newapi/ooniapi/measurements.py

```diff
@@ -367,6 +495,103 @@ def get_raw_measurement():
     return resp
 
 
+def _get_measurement_meta_postgresql(
+    report_id: str, input_
+) -> dict:  # pragma: no cover
+    # Given report_id + input, fetch measurement data from fastpath table
+    query = """SELECT
+        anomaly,
+        confirmed,
+        msm_failure AS failure,
+        input,
+        measurement_start_time,
+        probe_asn,
+        probe_cc,
+        report_id,
+        measurement_uid,
+        CAST(scores AS varchar) AS scores,
+        test_name,
+        test_start_time
+    """
+    # fastpath uses input = '' for empty values
+    if input_ is None:
+        query += """
+        FROM fastpath
+        WHERE fastpath.report_id = :report_id
+        AND (fastpath.input IS NULL or fastpath.input = '')
+        AND probe_asn != 0
+        """
+    else:
+        query += """
+            , citizenlab.category_code AS category_code
+        FROM fastpath
+        LEFT OUTER JOIN citizenlab ON citizenlab.url = fastpath.input
+        WHERE fastpath.input = :input
+        AND fastpath.report_id = :report_id
+        AND probe_asn != 0
+        """
+    query_params = dict(input=input_, report_id=report_id)
+    q = current_app.db_session.execute(query, query_params)
+    msmt_meta = q.fetchone()
+    if msmt_meta is None:
+        return {}  # measurement not found
+
+    return dict(msmt_meta)
+
+
+@metrics.timer("get_measurement_meta_clickhouse")
+def _get_measurement_meta_clickhouse(report_id: str, input_) -> dict:
+    # Given report_id + input, fetch measurement data from fastpath table
+    query = "SELECT * FROM fastpath "
+    if input_ is None:
+        # fastpath uses input = '' for empty values
+        query += "WHERE report_id = :report_id AND input = '' "
+    else:
+        query += """
+        LEFT OUTER JOIN citizenlab ON citizenlab.url = fastpath.input
+        WHERE fastpath.input = :input
+        AND fastpath.report_id = :report_id
+        """
+    query_params = dict(input=input_, report_id=report_id)
+    try:
+        rid_t = datetime.strptime(report_id[:8], "%Y%m%d")
+        query_params["begin"] = rid_t - timedelta(days=30)
+        query_params["end"] = rid_t + timedelta(days=30)
+        query += """
+        AND fastpath.measurement_start_time > :begin
+        AND fastpath.measurement_start_time < :end
```

Comment:
It's worth noting that this will render measurements that are uploaded more than 30 days after their measurement upload time inaccessible.

We should probably note this as a comment and investigate if there is something else we can to do optimise this lookup.

Reply:
We want to restrict the range for performance reasons. Is there anything different we can do?

Reply:
Added comment. At this time we never encountered measurements that are not accessible due to this filtering (we are using a generous 60 day range).